### ISSUE TITLE
IndexHierarchy.from_pandas allows construction of bloated indices

### ISSUE THREAD

**Issue by chaburkland** [static-frame/static-frame#652]
**Description**

Pandas MultiIndex allows for bloated indices. We assumed incorrectly that pandas did NOT allow this.

We definitely need to update the `from_pandas` implementation to disallow bloat, but I'm wondering if we want to enforce this in init. Personally I don't want to pay that cost as it seems expensive, rather, continue to have the constructors enforce it. `from_pandas` is the only such constructor that I'm aware of.

**Example**

Provide an interactive Python session, or Python code, to demonstrate the issue. Please keep example data to a minimum.

```python

mi = pd.MultiIndex([tuple("ABC"), tuple("DEF")], [[0, 0, 1], [1,2,2]])

ih1 = sf.IndexHierarchy.from_pandas(mi)
ih2 = sf.IndexHierarchy.from_values_per_depth([tuple("AAB"), tuple("EFF")])

assert ih1.equals(ih2)

ih1.loc[ih2] # This works as expected
ih2.loc[ih1] # This raises a KeyError.
```

**Platform**
```python
<Series: platform>
<Index>
platform           Linux-5.4.0-128-generic-x86_64-with-glibc2.29
sys.version        3.8.12 (default, Aug 26 2022, 13:23:08) [GCC 9.4.0]
static-frame       0.9.17
numpy              1.19.2
pandas             0.25.3
xlsxwriter         1.1.5
openpyxl           3.0.9
xarray             <ModuleNotFoundError>
tables             3.6.1
pyarrow            0.17.0
msgpack            (1, 0, 0)
msgpack_numpy      None
<<U13>             <object>
```

**Comment by flexatone** [comment]
Thanks for posting this issue. I think we need to update `from_pandas` to not use Pandas potentially "bloated" internal representation, and instead just take an array  per depth level and use our new `from_values_per_depth` constructor.

@chaburkland : Please let me know if you would like to explore this change, otherwise I am happy to do it.

**Comment by chaburkland** [comment]
I'll work on this with @michaeljpeters . Also, I found a method from pandas that will allow the code change to be very minimal: https://pandas.pydata.org/docs/reference/api/pandas.MultiIndex.remove_unused_levels.html

**Comment by flexatone** [comment]
This was corrected with the merge of #653 .
